### PR TITLE
feat(Booyah): add presence

### LINKED ISSUE #5677: Booyah | booyah.live
### Website name

Booyah

### Website URL

https://booyah.live/

### Website logo

https://www.google.com/url?sa=i&url=https%3A%2F%2Fbooyah.live%2Fchannels%2F396080&psig=AOvVaw0-_PCJD6kKVejF72jz7GxC&ust=1643407898044000&source=images&cd=vfe&ved=0CAsQjRxqFwoTCOi1tdP50vUCFQAAAAAdAAAAABAD

### Prerequisites

- [ ] It is a paid service
- [ ] It displays NSFW content
- [ ] It is region restricted

### Description

Booyah is a plataform like twitch. I'd like the platform to be configured the same as twitch. That all :D

### PR DESCRIPTION
**Pull request by AndersonPGS**
**Added Booyah Website**
https://booyah.live/

Resolves #5677

### CODE REVIEW

**Comment by Isa27-0** [comment]
Issue to be resolved: #5677

File: websites/B/Booyah/presence.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+				live: "general.live",
+				viewHome: "general.viewHome",
+				browse: "general.browsing",
+				watchingLive: "general.watchingLive",
+				watchingVid: "general.watchingVid",
+				viewCategory: "general.viewCategory",
+				viewTheir: "twitch.viewTheir",
+				channelSettings: "twitch.channelSettings",
+				dashboard: "twitch.dashboard",
+				viewFollow: "twitch.viewFollow",
```
Comment (Bas950) [inline websites/B/Booyah/presence.ts:25]: use twitch strings will need you to update the language setting accordingly. https://docs.premid.app/en/dev/presence/metadata#multilanguage

**Review by Bas950 — DISMISSED** [review]

File: websites/B/Booyah/presence.ts (hunk 1 above)
Comment (AndersonPGS) [reply websites/B/Booyah/presence.ts:25]: I added twitch in multilanguage.

File: websites/B/Booyah/presence.ts [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+			delete presenceData.smallImageText;
+			presenceData.details = strings.viewTheir;
+			presenceData.state = strings.channelSettings;
+			if (pathname.includes("/dashboard"))
+				presenceData.state = strings.dashboard;
+		}
+	} else if (pathname.startsWith("/vods") || (streamer && title)) {
+		if (vidTimer) {
+			let paused = false;
+			if (document.querySelector(".playback-btn") !== null) paused = true;
```
Comment (Slowlife01) [inline websites/B/Booyah/presence.ts:141]:
```suggestion
			let paused = document.querySelector(".playback-btn") !== null;
```

File: websites/B/Booyah/presence.ts [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+			delete presenceData.smallImageKey;
+			delete presenceData.smallImageText;
+			presenceData.details = strings.viewTheir;
+			presenceData.state = strings.channelSettings;
+			if (pathname.includes("/dashboard"))
+				presenceData.state = strings.dashboard;
+		}
+	} else if (pathname.startsWith("/vods") || (streamer && title)) {
+		if (vidTimer) {
+			let paused = document.querySelector(".playback-btn") !== null;
```
Comment (ImRodry) [inline websites/B/Booyah/presence.ts:140]:
```suggestion
			let paused = !!document.querySelector(".playback-btn");
```

File: websites/B/Booyah/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+		"name": "Anderson Paiva",
+		"id": "584828331954929909"
+	},
+	"service": "Booyah",
+	"description": {
+		"en": "BOOYAH! is a leading livestreaming and community platform designed for gaming enthusiasts, powered by Garena."
+	},
+	"url": "booyah.live",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/Iv1PUYy.jpeg",
```
Comment (ImRodry) [inline websites/B/Booyah/dist/metadata.json:13]: Must be 512x at least

**Review by ImRodry — CHANGES_REQUESTED** [review]

File: websites/B/Booyah/presence.ts [hunk 5]
```diff
@@ -0,0 +1,214 @@
+let elapsed = Math.floor(Date.now() / 1000),
+	oldLang = "en",
+	prevUrl = document.location.href;
+
+const presence = new Presence({
+		clientId: "943158612090568745"
+	}),
+	getElement = (query: string): string | undefined => {
+		return document.querySelector(query)?.textContent;
+	},
```
[the remark this answers is not in the document]
Comment (AndersonPGS) [reply websites/B/Booyah/presence.ts:10]: How could I do this?

File: websites/B/Booyah/dist/metadata.json (hunk 4 above)
Comment (AndersonPGS) [reply websites/B/Booyah/dist/metadata.json]: Ready

File: websites/B/Booyah/presence.ts [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+			profilePic,
+			buttons
+		] = await Promise.all([
+			presence.getSetting<boolean>("timestamp"),
+			presence.getSetting<string>("lang").catch(() => "en"),
+			presence.getSetting<boolean>("privacy"),
+			presence.getSetting<string>("vidDetail"),
+			presence.getSetting<string>("vidState"),
+			presence.getSetting<string>("streamDetail"),
+			presence.getSetting<string>("streamState"),
```
[the remark this answers is not in the document]
Comment (AndersonPGS) [reply websites/B/Booyah/presence.ts:61]: in my head it seems necessary

File: websites/B/Booyah/presence.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (AndersonPGS) [reply websites/B/Booyah/presence.ts]: removed

File: websites/B/Booyah/presence.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (EncryptedDev) [reply websites/B/Booyah/presence.ts:58]: If you want details on one; you would likely want details on the other - same for state. Simplfy this to make a details input and state input.

**Comment by EncryptedDev** [comment]
Resolve conversations once you have acknowledged and acted on them.

File: websites/B/Booyah/presence.ts [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+	else if (pathname.startsWith("/games")) {
+		presenceData.smallImageKey = "reading";
+		presenceData.smallImageText = strings.browse;
+		presenceData.details = strings.browse;
+		if (!privacy) {
+			presenceData.details = strings.viewCategory;
+			if (
+				document.querySelector(
+					"#layout-content > div > div.views-game-info > div.game-info > div.game-name"
+				) !== null
```
Comment (EncryptedDev) [inline websites/B/Booyah/presence.ts:92]:
```suggestion
				document.querySelector(
					"#layout-content > div > div.views-game-info > div.game-info > div.game-name"
				)
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/B/Booyah/presence.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (AndersonPGS) [reply websites/B/Booyah/presence.ts]: Ok

**Review by Slowlife01 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**